src/TLAuctionHouse.sol

```diff
@@ -0,0 +1,563 @@
+        _setWethAddress(newWethAddress);
+
+        emit WethUpdated(prevWeth, newWethAddress);
+    }
+
+    /// @notice Function to set the min bid increase settings
+    /// @dev Requires owner
+    /// @param newMinBidIncreasePerc The new minimum bid increase nominal percentage, out of `BASIS`
+    /// @param newMinBidIncreaseLimit The new minimum bid increase absolute limit
+    function setMinBidIncreaseSettings(uint256 newMinBidIncreasePerc, uint256 newMinBidIncreaseLimit)
+        external
+        onlyOwner
+    {
+        _setMinBidIncreaseSettings(newMinBidIncreasePerc, newMinBidIncreaseLimit);
+    }
+
+    /// @notice Function to set the protocol fee settings
+    /// @dev Requires owner
+    /// @param newProtocolFeeReceiver The new protocol fee receiver
+    /// @param newProtocolFeePerc The new protocol fee percentage, out of `BASIS`
+    /// @param newProtocolFeeLimit The new protocol fee limit
+    function setProtocolFeeSettings(
+        address newProtocolFeeReceiver,
+        uint256 newProtocolFeePerc,
+        uint256 newProtocolFeeLimit
+    ) external onlyOwner {
+        _setProtocolFeeSettings(newProtocolFeeReceiver, newProtocolFeePerc, newProtocolFeeLimit);
+    }
+
+    /// @notice Function to pause the contract
+    /// @dev Requires owner
+    /// @param status The boolean to set the internal pause variable
+    function pause(bool status) external onlyOwner {
+        if (status) {
+            _pause();
+        } else {
+            _unpause();
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                            Auction Configuration Functions
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Function to configure an auction
+    /// @dev Requires the following items to be true
+    ///     - contract is not paused
+    ///     - the auction hasn't been configured yet
+    ///     - msg.sender is the owner of the token
+    ///     - auction house is approved for all
+    ///     - payoutReceiver isn't the zero address
+    /// @param nftAddress The nft contract address
+    /// @param tokenId The nft token id
+    /// @param payoutReceiver The address that receives the payout from the auction
+    /// @param currencyAddress The currency to use
+    /// @param reservePrice The auction reserve price
+    /// @param auctionOpenTime The time at which bidding is allowed
+    /// @param duration The duration of the auction after it is started
+    function configureAuction(
+        address nftAddress,
+        uint256 tokenId,
+        address payoutReceiver,
+        address currencyAddress,
+        uint256 reservePrice,
+        uint256 auctionOpenTime,
+        uint256 duration
+    ) external whenNotPaused {
+        IERC721 nft = IERC721(nftAddress);
+        Auction memory auction = _auctions[nftAddress][tokenId];
+
+        if (auction.seller != address(0)) revert AuctionAlreadyConfigured();
```

Comment:
should update to check to see if the auction exists from the previous token owner and not require the new owner to cancel and then configure

Reply:
agreed